### DISCUSSION TITLE
Emit event by sender account but show account address `0x0`

### DISCUSSION THREAD

**Discussion by spielcrypto** [aptos-labs/aptos-developer-discussions#345] [Questions] [answered]
### Discord user ID

spielcrypto

### Describe your question in detail.

I have this struct event in my smart contract:
```rust
#[event]
    struct BurnCoinEvent has drop, store {
        sender: address,
        coin: String,
        symbol: String,
    }
```

and after in this function I emit it:
```rust
public entry fun coins<CoinType, BurnCoinType>(account: &signer, nfts_required_holding: vector<address>) acquires Config {
        checkHolderOrPayFees<CoinType>(account, nfts_required_holding);
        let account_address = signer::address_of(account);

        aptos_account::transfer_coins<BurnCoinType>(account, @ghostly_addr, coin::balance<BurnCoinType>(account_address));
        
        let burn_coin_event = BurnCoinEvent {
            sender: account_address,
            coin: coin::name<CoinType>(),
            symbol: coin::symbol<CoinType>()
        };

        event::emit(burn_coin_event);
    }
```

when some account call the function coins the event is emitted but with account `0x0`, [here an example](https://explorer.aptoslabs.com/txn/5519062748/events?network=testnet) in the event type `
0x9ed8aa766c59553a345523a1840e4727af9e16d6b9b98c0defdea2f639bc0e63::burn::BurnCoinEvent`
In that I don't see way to retrieve that event from the sender account because if request with this curl:
```cmd
 curl --request GET \                                                                 ✔  6s 
  --url https://api.testnet.aptoslabs.com/v1/accounts/0x5cf7605e9a39a8a0610b831ef1bd0d4533b3c97bbf8b94d901437afe100980ce/events/0x9ed8aa766c59553a345523a1840e4727af9e16d6b9b98c0defdea2f639bc0e63::burn::BurnCoinEvent/amount \
  --header 'Accept: application/json, application/x-bcs'
```
I get `Resource not found by Address(0x5cf7605e9a39a8a0610b831ef1bd0d4533b3c97bbf8b94d901437afe100980ce)`

### Which operating system are you using?

Linux (Ubuntu, Fedora, Windows WSL, etc.)

### Which SDK or tool are you using? (if any)

Indexer API

**Comment by adityaaggarwal12** [comment]
Hey @dancespiele, Hope you're doing well.

As far as I know, there are two types of events in Aptos:
1. Module events
2. Event Handle Events (deprecated)

"Account Address" and "Sequence Number" were part of legacy Event Handle Events, where each event had a unique GUID - "Event Handle" and a sequence number.

Currently, only legacy Event Handle Events are supported in node APIs.

New Module Events do not have a GUID, Account address, or sequence number. These are all set to zero due to backwards compatibility.

To access module events, you can use Aptos's indexer GraphQL:
`https://cloud.hasura.io/public/graphiql?endpoint=https://api.testnet.aptoslabs.com/v1/graphql`

Here's an example query for your event:

```graphql
query MyQuery {
  events(
    where: {transaction_version: {_eq: "5519062748"}, type: {_eq: "0x9ed8aa766c59553a345523a1840e4727af9e16d6b9b98c0defdea2f639bc0e63::burn::BurnCoinEvent"}}
  ) {
    account_address
    creation_number
    event_index
    indexed_type
    transaction_block_height
    transaction_version
    type
    sequence_number
    data
  }
}
```

**Reply by adityaaggarwal12** [reply]
found this docs link - `https://aptos.dev/en/network/blockchain/events` you can refer to this for more detailed info

**Reply by spielcrypto** [reply]
thank you for your reply, and that query works but if in that query I remove `transactions_version` in order to get all the transactions that comes from that type I get this error:
```json
{
  "errors": {
    "message": "Unexpected token 'R', \"Request Ti\"... is not valid JSON",
    "stack": "SyntaxError: Unexpected token 'R', \"Request Ti\"... is not valid JSON"
  }
}
```
Which is really what I need but doesn't work

**Reply by spielcrypto** [reply, marked as answer]
ok if I write `indexed_type` instead of `type` works and I have what I want. Thanks!
